### DISCUSSION TITLE
how to save bmp image as grayscale image (8bit)

### DISCUSSION THREAD

**Discussion by Bestbenben** [nothings/stb#1357] [Q&A] [answered]
const char* sourceImagePath = "d:/src.bmp";
	const char* destImagePath = "d:/res.bmp";
	int iw = 0, ih = 0;
	unsigned char *idata = stbi_load(sourceImagePath, &iw, &ih, 0, 1);
	stbi_write_bmp(destImagePath, iw, ih, 1, idata);
	stbi_image_free(idata);

i am unsuccessful in saving an image as an 8bit grayscale image

**Comment by nothings** [comment, marked as answer]
Yes, this is documented behavior.

"The BMP format expands Y to RGB in the file format and does not output alpha." The expected result is an RGB image, although the results itself should be a grayscale RGB file.

**Reply by Bestbenben** [reply]
Thank you.
